cli/azd/pkg/account/manager.go

```diff
@@ -0,0 +1,230 @@
+package account
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/azure/azure-dev/cli/azd/pkg/config"
+	"github.com/azure/azure-dev/cli/azd/pkg/tools/azcli"
+	"golang.org/x/exp/slices"
+)
+
+// Manages azd account configuration
+// `az cli` is not required and will only be called `azd` default have not already been set.
+type Manager struct {
+	azCli azcli.AzCli
+}
+
+// Creates a new Account Manager instance
+func NewManager(ctx context.Context) *Manager {
+	return &Manager{
+		azCli: azcli.GetAzCli(ctx),
+	}
+}
+
+// Gets the default subscription for the logged in account.
+// 1. Returns AZD config defaults if exists
+// 2. Returns AZ CLI defaults if exists
+func (m *Manager) GetAccountDefaults(ctx context.Context) (*config.Config, error) {
```

Comment:
env vars will still be supported in projects which is where this helps.  Having env vars for top level config values doesn't feel like it would add much value here and wouldn't need to be set in CI or similar.